src/SFA.DAS.EAS.Portal.UnitTests/Worker/EventHandlers/Commitments/CohortApprovalRequestedEventHandlerTests.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Threading;
+using System.Threading.Tasks;
+using Moq;
+using NServiceBus;
+using NUnit.Framework;
+using SFA.DAS.CommitmentsV2.Messages.Events;
+using SFA.DAS.EAS.Portal.Application.Adapters;
+using SFA.DAS.EAS.Portal.Application.Commands;
+using SFA.DAS.EAS.Portal.Application.Commands.Cohort;
+using SFA.DAS.EAS.Portal.Application.Services;
+using SFA.DAS.EAS.Portal.UnitTests.Builders;
+using SFA.DAS.EAS.Portal.Worker.EventHandlers.Commitments;
+
+namespace SFA.DAS.EAS.Portal.UnitTests.Worker.EventHandlers.Commitments
+{
+    [Parallelizable]
+    [TestFixture]
+    public class CohortApprovalRequestedEventHandlerTests
+    {
+        public class TestContext
+        {
+            public CohortApprovalRequestedByProviderEventHandler Sut { get; private set; }
+            public Mock<ICommandHandler<CohortApprovalRequestedCommand>> MockHandler { get; private set; }
+            public Mock<IAdapter<CohortApprovalRequestedByProvider, CohortApprovalRequestedCommand>> MockAdapter { get; private set; }
+            public Mock<IMessageHandlerContext> MockMessageHandlerContext { get; private set; }
+            public Mock<IMessageContext> MockMessageContext { get; private set; }
+
+            public TestContext()
+            {
+                MockHandler = new Mock<ICommandHandler<CohortApprovalRequestedCommand>>();
+                MockAdapter = new Mock<IAdapter<CohortApprovalRequestedByProvider, CohortApprovalRequestedCommand>>();
+                MockMessageContext = new Mock<IMessageContext>();
+                MockMessageHandlerContext = new Mock<IMessageHandlerContext>();
+                MockMessageHandlerContext
+                    .Setup(m => m.MessageHeaders)
+                    .Returns(
+                    new Dictionary<string, string>
+                    {
+                        { "NServiceBus.TimeSent", DateTime.Now.ToUniversalTime().ToString("yyyy-MM-dd HH:mm:ss:ffffff Z", CultureInfo.InvariantCulture) }
+                    });
+
+                Sut = new CohortApprovalRequestedByProviderEventHandler(MockHandler.Object, MockAdapter.Object, MockMessageContext.Object);
+            }
+        }
+
+        public class Handle : CohortApprovalRequestedEventHandlerTests
+        {
+            [Test]
+            public async Task WhenCalled_ThenTheAdapterIsCalledToConvertTheMessage()
```

Comment:
This unit test is testing an implementation detail of the code under test. Unit tests should test any returned values are correct and any expected external actions are verified. By testing an implementation detail, the test can fail if the code under test is refactored, but is still correct. Whereas we want the test to continue to pass if the code under test is behaving correctly.

So for testing the handler, we should test that the handler is called with the result of the conversion, which is what the next test does 👍.